pkg/binary/envoy/debug/pii_test.go

```diff
@@ -0,0 +1,128 @@
+		" \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"" +
+		" %UPSTREAM_CLUSTER% %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS% %REQUESTED_SERVER_NAME%\n"
+
+	formats, ok := shell.Split(format)
+	if !ok {
+		t.Fatalf("error in splitting the format: %s", format)
+	}
+
+	validLog := "[2019-09-05T17:53:36.908Z] \"GET /productpage HTTP/1.1\" 200" +
+		" - \" - \" \" - \" 0 5183 47 46 \"10.28.2.1\" \"Mozilla/5.0 (X11; Linux x86_64)" +
+		" AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36\"" +
+		" \"4273c09c-2534-962a-b7da-04a04cf39e88\" \"35.199.144.198\" \"127.0.0.1:9080\"" +
+		" inbound|9080|http|productpage.default.svc.cluster.local - 10.28.2.27:9080 10.28.2.1:0" +
+		" outbound_.9080_._.productpage.default.svc.cluster.local"
+	logWithOneField := `[2019-09-05T17:53:36.908Z]`
+
+	type Out struct {
+		logs []string
+		err  error
+	}
+	// test cases wrapped up in a map where key is the test case name and value is the struct
+	// defining the inputs and expecte outcome
+	tests := map[string]struct {
+		logs        []string
+		format      string
+		containsPII map[string]bool
+		want        Out
+	}{
+		// positive test cases
+		// only return an array containing one valid log with [%START_TIME%] field hashed
+		"simple": {logs: []string{validLog, logWithOneField},
+			format: format, containsPII: map[string]bool{"[%START_TIME%]": true},
+			want: Out{logs: []string{validLog}, err: nil},
+		},
+		"multiple_pii_fields": {logs: []string{validLog, logWithOneField},
+			format: format, containsPII: map[string]bool{"[%START_TIME%]": true,
+				"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)% %PROTOCOL%": true, "%RESPONSE_CODE%": true},
+			want: Out{logs: []string{validLog}, err: nil},
+		},
+
+		// negative test cases
+		"empty_logs": {logs: []string{},
+			format: format, containsPII: map[string]bool{"[%START_TIME%]": true},
+			want: Out{logs: []string{}, err: nil},
+		},
+		"all_invalid_logs": {logs: []string{"test", "test1", "test2"},
+			format: format, containsPII: map[string]bool{"[%START_TIME%]": true},
+			want: Out{logs: []string{}, err: nil},
+		},
+		"empty_format_string": {logs: []string{validLog, logWithOneField},
+			format: "", containsPII: map[string]bool{"[%START_TIME%]": true},
+			want: Out{logs: []string{}, err: nil},
+		},
+		"empty_pii_map": {logs: []string{validLog, logWithOneField},
+			format: format, containsPII: map[string]bool{},
+			want: Out{logs: []string{validLog}, err: nil},
+		},
+		"invalid_pii_map": {logs: []string{validLog, logWithOneField},
+			format: format, containsPII: map[string]bool{"test": true, "test2": true},
+			want: Out{logs: []string{validLog}, err: nil},
+		},
+	}
+
+	// defining subtest with t.Run
+	for name, tc := range tests {
+		logs := tc.logs
+		format := tc.format
+		containsPII := tc.containsPII
+		expectedLogs := tc.want.logs
+		expectedErr := tc.want.err
```

Comment:
Don't do this, just use the fields directly. This is why you gave the fields descriptive names. Instead you should rename `tc` to something more useful. Either `case`, so the text reads naturally (`actual, err := ProcessLogs(case.in, case.format, case.pii)`) or `c` by convention (mirroring `t`).